src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,448 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity >=0.8.0;
+
+import "./TokenChecks.sol";
+
+/** 
+ * @title Token Fuzz Checks
+ * @dev Contains fuzz checks for token contracts similar to test/SampleToken.sol
+ */
+contract TokenFuzzChecks is TokenChecks {
+
+    using GodMode for *;
+
+    // ************************************************************************************************************
+    // Mint/Burn
+    // ************************************************************************************************************
+
+    function fuzzBulkCheckMintBurn(
+        address _token,
+        string memory _contractName,
+        address who,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) internal {
+        fuzzCheckMint(_token, _contractName, who, mintAmount);
+        fuzzCheckBurn(_token, who, mintAmount, burnAmount);
+        fuzzCheckBurnInsufficientBalance(_token, _contractName, who, mintAmount, burnAmount);
+        fuzzCheckTokenModifiers(_token, _contractName, who);
+    }
+
+    function fuzzCheckMint(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount
+    ) internal {
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply);
+        uint256 prevWard = TokenLike(_token).wards(address(this));
+        _token.setWard(address(this), 1);
+        if (to != address(0) && to != _token) {
+            vm.expectEmit(true, true, true, true);
+            emit Transfer(address(0), to, mintAmount);
+        } else {
+            vm.expectRevert(abi.encodePacked(_contractName, "/invalid-address"));
+        }
+
+        TokenLike(_token).mint(to, mintAmount);
+
+        if (to != address(0) && to != _token) {
+            assertEq(TokenLike(_token).totalSupply(), prevSupply + mintAmount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + mintAmount);
+        }
+        _token.setWard(address(this), prevWard);
+    }
+
+    function fuzzCheckBurn(
```

Comment:
One final and minor thing, can we call all of these functions `checkFuzz...` instead of `fuzzCheck...`?
That way we would keep better consistency with the regular tests/checks.

Reply:
Done